src/components/structures/login/Registration.js

```diff
@@ -135,154 +128,123 @@ module.exports = React.createClass({
+            // retrieve team config and see welcome page etc.
+            this._rtsClient.trackReferral(
+                this.props.referrer || '', // Default to empty string = not referred
+                this.registerLogic.params.idSid,
+                this.registerLogic.params.clientSecret
+            ).then((data) => {
+                const teamToken = data.team_token;
+                // Store for use /w welcome pages
+                window.localStorage.setItem('mx_team_token', teamToken);
+                this.props.onTeamMemberRegistered(teamToken);
+
+                this._rtsClient.getTeam(teamToken).then((team) => {
+                    console.log(
+                        `User successfully registered with team ${team.name}`
+                    );
+                    if (!team.rooms) {
+                        return;
+                    }
+                    // Auto-join rooms
+                    team.rooms.forEach((room) => {
+                        if (room.auto_join && room.room_id) {
+                            console.log(`Auto-joining ${room.room_id}`);
+                            MatrixClientPeg.get().joinRoom(room.room_id);
                         }
-                        // Auto-join rooms
-                        team.rooms.forEach((room) => {
-                            if (room.auto_join && room.room_id) {
-                                console.log(`Auto-joining ${room.room_id}`);
-                                MatrixClientPeg.get().joinRoom(room.room_id);
-                            }
-                        });
-                    }, (err) => {
-                        console.error('Error getting team config', err);
                     });
 
                     return teamToken;
                 }, (err) => {
-                    console.error('Error tracking referral', err);
+                    console.error('Error getting team config', err);
                 });
-            }
-
-            return trackPromise.then((teamToken) => {
-                console.info('Team token promise',teamToken);
-                self.props.onLoggedIn({
-                    userId: response.user_id,
-                    deviceId: response.device_id,
-                    homeserverUrl: self.registerLogic.getHomeserverUrl(),
-                    identityServerUrl: self.registerLogic.getIdentityServerUrl(),
-                    accessToken: response.access_token
-                }, teamToken);
-            }).then(() => {
-                self._setupPushers();
-            });
-        }, function(err) {
-            if (err.message) {
-                self.setState({
-                    errorText: err.message
-                });
-            }
-            self.setState({
-                busy: false
+            }, (err) => {
+                console.error('Error tracking referral', err);
             });
-            console.log(err);
-        });
+        }
+
+        trackPromise.then((teamToken) => {
```

Comment:
Thanks - yeah, this was git making a mess of merging in luke's PR, and this must have been something I lost in trying to fix the merge.